### PR TITLE
People 162 enable leader to see teammates skills

### PR DESCRIPTION
**Pull request by elharion**
# Ticket
https://netguru.atlassian.net/browse/PEOPLE-162

# Description
- Enable leaders to see skills of his teammates.

### Before merge checklist
- [x] CircleCI is passing
- [ ] Proper labels are set

### CODE REVIEW

File: spec/factories/team.rb [hunk 1]
```diff
@@ -1,5 +1,14 @@
 FactoryGirl.define do
   factory :team do
     name { Faker::Name.name }
+
+    factory :team_with_members
+      transient do
+        users_count 8
+      end
+
+      after(:create) do |team, evaluator|
```
Comment (netguru-hound) [inline spec/factories/team.rb]: Inconsistent indentation detected.

File: spec/factories/team.rb [hunk 2]
```diff
@@ -1,5 +1,14 @@
 FactoryGirl.define do
   factory :team do
     name { Faker::Name.name }
+
+    factory :team_with_members
+      transient do
```
Comment (netguru-hound) [inline spec/factories/team.rb]: Inconsistent indentation detected.

File: app/policies/application_policy.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    @user = user
+    @record = record
+  end
+
+  def index?
+    false
+  end
+
+  def show?
+    scope.where(:id => record.id).exists?
```
Comment (netguru-hound) [inline app/policies/application_policy.rb]: Use the new Ruby 1.9 hash syntax.

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -30,6 +33,10 @@ def connect_github
 
   private
 
+  def user_not_authorized
+    redirect_to((request.referer || root_path), alert: 'You are not authorized to access this section.')
```
Comment (netguru-hound) [inline app/controllers/application_controller.rb]: Line is too long. [104/100]

Comment (elharion) [inline app/controllers/application_controller.rb]: meh

File: app/views/layouts/_header.haml [hunk 5]
```diff
@@ -16,7 +16,19 @@
...
+              Skills
+              %b.caret
+            %ul.dropdown-menu
+              %li{ class: menu_class('skills')}= link_to 'My skills', user_skill_rates_path
+              %li{ class: menu_class('skills')}= link_to 'My skills history', skills_history_user_path(current_user)
+              - if current_user.leader?
+                %li{ class: 'dropdown-header' } My team members
+                %li{ class: 'divider', role: 'separator' }
+                - team_members.each do |member|
+                  %li{ class: menu_class('skills') }= link_to member.name, skills_history_user_path(member)
```
Comment (netguru-hound) [inline app/views/layouts/_header.haml]: Line is too long. [107/80]

File: app/views/layouts/_header.haml [hunk 6]
```diff
@@ -16,7 +16,19 @@
...
+          %li.dropdown
+            %a.dropdown-toggle{ href: '#', 'data-toggle' => 'dropdown' }
+              Skills
+              %b.caret
+            %ul.dropdown-menu
+              %li{ class: menu_class('skills')}= link_to 'My skills', user_skill_rates_path
+              %li{ class: menu_class('skills')}= link_to 'My skills history', skills_history_user_path(current_user)
+              - if current_user.leader?
+                %li{ class: 'dropdown-header' } My team members
+                %li{ class: 'divider', role: 'separator' }
```
Comment (netguru-hound) [inline app/views/layouts/_header.haml]: Avoid defining `class` in attributes hash for static class names

File: app/views/layouts/_header.haml [hunk 7]
```diff
@@ -16,7 +16,19 @@
...
-          %li{ class: menu_class('skill_user_rates') }= link_to "Skills", user_skill_rates_path
+          %li.dropdown
+            %a.dropdown-toggle{ href: '#', 'data-toggle' => 'dropdown' }
+              Skills
+              %b.caret
+            %ul.dropdown-menu
+              %li{ class: menu_class('skills')}= link_to 'My skills', user_skill_rates_path
+              %li{ class: menu_class('skills')}= link_to 'My skills history', skills_history_user_path(current_user)
+              - if current_user.leader?
+                %li{ class: 'dropdown-header' } My team members
```
Comment (netguru-hound) [inline app/views/layouts/_header.haml]: Avoid defining `class` in attributes hash for static class names

File: app/views/layouts/_header.haml [hunk 8]
```diff
@@ -16,7 +16,19 @@
...
           %li{ class: menu_class('teams') }= link_to "Teams", teams_path
           %li{ class: menu_class('project_info') }= link_to "Project Info", project_info_index_path
-          %li{ class: menu_class('skill_user_rates') }= link_to "Skills", user_skill_rates_path
+          %li.dropdown
+            %a.dropdown-toggle{ href: '#', 'data-toggle' => 'dropdown' }
+              Skills
+              %b.caret
+            %ul.dropdown-menu
+              %li{ class: menu_class('skills')}= link_to 'My skills', user_skill_rates_path
+              %li{ class: menu_class('skills')}= link_to 'My skills history', skills_history_user_path(current_user)
```
Comment (netguru-hound) [inline app/views/layouts/_header.haml]: Line is too long. [116/80]<br>Hash attribute should end with one space before the closing brace

File: app/views/layouts/_header.haml [hunk 9]
```diff
@@ -16,7 +16,19 @@
           %li{ class: menu_class('dashboard') }= link_to "Projects", dashboard_index_path
           %li{ class: menu_class('teams') }= link_to "Teams", teams_path
           %li{ class: menu_class('project_info') }= link_to "Project Info", project_info_index_path
-          %li{ class: menu_class('skill_user_rates') }= link_to "Skills", user_skill_rates_path
+          %li.dropdown
+            %a.dropdown-toggle{ href: '#', 'data-toggle' => 'dropdown' }
+              Skills
+              %b.caret
+            %ul.dropdown-menu
+              %li{ class: menu_class('skills')}= link_to 'My skills', user_skill_rates_path
```
Comment (netguru-hound) [inline app/views/layouts/_header.haml]: Line is too long. [91/80]<br>Hash attribute should end with one space before the closing brace

File: app/controllers/application_controller.rb [hunk 10]
```diff
@@ -45,6 +66,6 @@ def message_to_js
       @flashMessage[name] << msg
     end
     gon.rabl template: 'app/views/layouts/flash.rabl', as: 'flash'
-    flash.clear
+    # flash.clear
```
Comment (elharion) [inline app/controllers/application_controller.rb]: i don't think this line is necessary. Need to check whether there will be duplicated flash messages with js. So far seen none.

File: app/views/layouts/_header.haml [hunk 11]
```diff
@@ -16,7 +16,19 @@
...
           %li{ class: menu_class('teams') }= link_to "Teams", teams_path
           %li{ class: menu_class('project_info') }= link_to "Project Info", project_info_index_path
-          %li{ class: menu_class('skill_user_rates') }= link_to "Skills", user_skill_rates_path
+          %li.dropdown
+            %a.dropdown-toggle{ href: '#', 'data-toggle' => 'dropdown' }
+              Skills
+              %b.caret
+            %ul.dropdown-menu
+              %li{ class: menu_class('skills') }= link_to 'My skills', user_skill_rates_path
+              %li{ class: menu_class('skills') }= link_to 'My skills history', skills_history_user_path(current_user)
```
Comment (netguru-hound) [inline app/views/layouts/_header.haml]: Line is too long. [117/80]

File: app/views/layouts/_header.haml [hunk 12]
```diff
@@ -16,7 +16,19 @@
           %li{ class: menu_class('dashboard') }= link_to "Projects", dashboard_index_path
           %li{ class: menu_class('teams') }= link_to "Teams", teams_path
           %li{ class: menu_class('project_info') }= link_to "Project Info", project_info_index_path
-          %li{ class: menu_class('skill_user_rates') }= link_to "Skills", user_skill_rates_path
+          %li.dropdown
+            %a.dropdown-toggle{ href: '#', 'data-toggle' => 'dropdown' }
+              Skills
+              %b.caret
+            %ul.dropdown-menu
+              %li{ class: menu_class('skills') }= link_to 'My skills', user_skill_rates_path
```
Comment (netguru-hound) [inline app/views/layouts/_header.haml]: Line is too long. [92/80]

File: app/react/components/rate-scale.jsx [hunk 13]
```diff
@@ -77,7 +77,7 @@ export default class RateScale extends React.Component {
         ></i>
       </li>
     ) : (
-      ''
+      <li />
```
Comment (elharion) [inline app/react/components/rate-scale.jsx]: this fixes wrong line-height on starts.

File: app/controllers/application_controller.rb (hunk 10 above)
Comment (DariuszMusielak) [inline app/controllers/application_controller.rb]: Please add some kind of TODO note

**Pull request merged**